dotCMS/src/main/java/com/dotcms/security/ContentSecurityPolicyUtil.java

```diff
@@ -0,0 +1,192 @@
+
+    private static boolean shouldCalculateNonce() {
+        final String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+        final PageMode pageMode = PageMode.get();
+        return pageMode == PageMode.LIVE && contentSecurityPolicyHeader.contains("nonce");
+    }
+
+    public static void addHeader(final HttpServletResponse response) {
+        if (isConfig()) {
+            String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+
+            for (final Entry<String, ContentSecurityPolicyResolver> entry : contentSecurityPolicyResolvers.entrySet()) {
+                if (contentSecurityPolicyHeader.contains(entry.getKey())) {
+                    contentSecurityPolicyHeader = entry.getValue().headerResolver.resolve(contentSecurityPolicyHeader);
+                }
+            }
+
+            response.addHeader("Content-Security-Policy", contentSecurityPolicyHeader);
+        }
+    }
+
+    public static String apply(final String htmlCodeParam) {
+        String htmlCodeResult = htmlCodeParam;
+
+        if (isConfig()) {
+            String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+
+            for (final Entry<String, ContentSecurityPolicyResolver> entry : contentSecurityPolicyResolvers.entrySet()) {
+                if (contentSecurityPolicyHeader.contains(entry.getKey())) {
+                    htmlCodeResult = entry.getValue().htmlCodeResolver
+                            .resolve(contentSecurityPolicyHeader, htmlCodeResult);
+                }
+            }
+        }
+
+        return htmlCodeResult;
+    }
+
+    private static String calculateNonceToScript(final String contentSecurityPolicyConfig,
+            final String htmlCode) {
+
+        final String nonce = getNonceFromCurrentRequest();
+        return htmlCode.replaceAll("<script", String.format("<script nonce='%s'",nonce));
+    }
+
+    private static String calculateNonceHeaderToScript(final String header) {
+        final String nonce = getNonceFromCurrentRequest();
+        return header.replace("{script-src nonce}", String.format("'nonce-%s'",nonce));
+    }
+
+    private static String calculateNonceToStyle(final String contentSecurityPolicyConfig,
+            final String htmlCode) {
+
+        final String nonce = getNonceFromCurrentRequest();
+        return htmlCode.replaceAll("<style", String.format("<style nonce='%s'",nonce));
+    }
+
+    private static String calculateNonceHeaderToStyle(final String header) {
+        final String nonce = getNonceFromCurrentRequest();
+        return header.replace("{style-src nonce}", String.format("'nonce-%s'",nonce));
+    }
+
+    private static String getNonceFromCurrentRequest() {
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        final String nonce = (String) request.getAttribute(NONCE_REQUEST_ATTRIBUTE);
+        return nonce;
+    }
+
+    @FunctionalInterface
+    interface HeaderResolver {
```

Comment:
If this is not public, you may replace with Function<String, String> and the HtmlCodeResolver with Function2<String, String, String>

Reply:
done https://github.com/dotCMS/core/pull/21337/commits/c8f185a995f8f27f3c0e841a257172acb0f7a94d#diff-28c06d2a8e3b90515eebb6eda4dd60f70d823d8164052d79e956963669cf0cb6R216